src/ompl/geometric/planners/aitstar/datastructures/Vertex.h

```diff
@@ -0,0 +1,371 @@
+                void removeFromReverseChildren(std::size_t vertexId);
+
+                /** \brief Returns this vertex's children in the reverse search tree. */
+                std::vector<std::shared_ptr<Vertex>> getReverseChildren() const;
+
+                /** \brief Whitelists a child. */
+                void whitelistAsChild(const std::shared_ptr<Vertex> &vertex) const;
+
+                /** \brief Returns whether a child is whitelisted. */
+                bool isWhitelistedAsChild(const std::shared_ptr<Vertex> &vertex) const;
+
+                /** \brief Blacklists a child. */
+                void blacklistAsChild(const std::shared_ptr<Vertex> &vertex) const;
+
+                /** \brief Returns whether a child is blacklisted. */
+                bool isBlacklistedAsChild(const std::shared_ptr<Vertex> &vertex) const;
+
+                /** \brief Returns whether the vertex knows its nearest neighbors on the current approximation. */
+                bool hasCachedNeighbors() const;
+
+                /** \brief Caches the neighbors for the current approximation. */
+                void cacheNeighbors(const std::vector<std::shared_ptr<Vertex>> &neighbors) const;
+
+                /** \brief Returns the nearest neighbors, throws if not up to date. */
+                const std::vector<std::shared_ptr<Vertex>> &getNeighbors() const;
+
+                /** \brief Registers that a child has been added to this vertex during the current forward search. */
+                void registerPoppedOutgoingEdgeDuringForwardSearch();
+
+                /** \brief Registers the expansion of this vertex during the current reverse search. */
+                void registerExpansionDuringReverseSearch();
+
+                /** \brief Unregisters the expansion of this vertex during the current reverse search, needed when a
+                 * reverse branch is invalidated due to a collision detection on an edge. */
+                void unregisterExpansionDuringReverseSearch();
+
+                /** \brief Registers the insertion of this vertex into the open queue during the current reverse
+                 * search. */
+                void registerInsertionIntoQueueDuringReverseSearch();
+
+                /** \brief Returns whether the vertex has had an outgoing edge popped during the current forward search.
+                 */
+                bool hasHadOutgoingEdgePoppedDuringCurrentForwardSearch() const;
+
+                /** \brief Returns whether the vertex has been expanded during the current reverse search. */
+                bool hasBeenExpandedDuringCurrentReverseSearch() const;
+
+                /** \brief Returns whether the vertex has been inserted into the queue during the current reverse
+                 * search. */
+                bool hasBeenInsertedIntoQueueDuringCurrentReverseSearch() const;
+
+                /** \brief Sets the reverse queue pointer of this vertex. */
+                void setReverseQueuePointer(
```

Comment:
maybe use `using FooHeapElement = ompl::BinaryHeap<....>::Element` somewhere to simplify this and the next method. Same comment applies to long BinaryHeap types elsewhere in this code.

Reply:
Sure, I can use a `using` declaration to help with the readability.